### DISCUSSION TITLE
dependency matcher not working

### DISCUSSION THREAD

**Discussion by Manish-ZS** [explosion/spaCy#11031] [Help: Coding & Implementations] [answered]
deppattern = [
        {"RIGHT_ID": "wrote", "RIGHT_ATTRS": {"LEMMA": "write"}},
        {"LEFT_ID": "wrote", "REL_OP": ">", "RIGHT_ID": "book", 
            "RIGHT_ATTRS": {"LEMMA": "book"}}
        ]

from spacy.matcher import DependencyMatcher

dmatcher = DependencyMatcher(nlp.vocab)

dmatcher.add("BOOK", [deppattern])

for _, (start, end) in dmatcher(doc):
    print(doc[start].sent)

error : [E098] Invalid pattern specified: expected both SPEC and PATTERN.

**Comment by polm** [comment, marked as answer]
Please read the [guide to Github Markdown for code](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code
). 

What version of spaCy are you using? This code runs without issue for me.

```
import spacy

nlp = spacy.load("en_core_web_sm")
doc = nlp("I wrote a book last year.")

deppattern = [
        {"RIGHT_ID": "wrote", "RIGHT_ATTRS": {"LEMMA": "write"}},
        {"LEFT_ID": "wrote", "REL_OP": ">", "RIGHT_ID": "book",
            "RIGHT_ATTRS": {"LEMMA": "book"}}
        ]

from spacy.matcher import DependencyMatcher

dmatcher = DependencyMatcher(nlp.vocab)

dmatcher.add("BOOK", [deppattern])

for _, (start, end) in dmatcher(doc):
    print(doc[start].sent)
```

**Reply by Manish-ZS** [reply]
i am using 2.3 version on kaggle
 it seems like its issue with kaggle docker which downgraded the spacy version from 3... .
 do you have 2.3 version code to do same thing?

**Reply by polm** [reply]
The DependencyMatcher was introduced in v3, so I'm surprised your code gets that far without an error - maybe it was added late in v2 as an experimental feature or something... I would strongly recommend figuring out how to upgrade your Kaggle environment.
